lib/msf/util/document_generator.rb

```diff
@@ -29,12 +29,16 @@ def self.spawn_module_document(mod, out_file)
       # @param mod [Msf::Module] Module to create document for.
       # @return [void]
       def self.get_module_document(mod)
-        md = ''
-
-        kb_path = File.join(PullRequestFinder::MANUAL_BASE_PATH, "#{mod.fullname}.md")
+        kb_path = nil
         kb = ''
 
-        if File.exist?(kb_path)
+        if File.exists?(File.join(PullRequestFinder::MANUAL_BASE_PATH, "#{mod.fullname}.md"))
+          kb_path = File.join(PullRequestFinder::MANUAL_BASE_PATH, "#{mod.fullname}.md")
+        elsif File.exists?(File.join(PullRequestFinder::USER_MANUAL_BASE_PATH, "#{mod.fullname}.md"))
```

Comment:
Should the user path take precedence over the system path?

Reply:
I think it should but then it'd be inconsistent with the ordering of the user logos, script and plugin directories as I noticed their usage in:
  * lib/msf/ui/console/command_dispatcher/resource.rb
  * lib/msf/ui/console/command_dispatcher/core.rb
  * lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb

Ideally, if there's a way to determine if a module was loaded from the `user_module_directory` then it'd search there first and vice versa but I couldn't find a way to do that.

Reply:
So looking into it a bit more thoroughly it looks like when [searching for plugins](https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/ui/console/command_dispatcher/core.rb#L803-L820), the user's directory has a higher precedence. I'm fine with which ever you think is best.

Reply:
According to @wvu-r7 we should have the user directory take precedence. The other option would be to combine the local docs with the msf docs if both exist.

Reply:
Sounds good, I'll get an update pushed to switch the search order.